src/main/java/com/mariamkatamashvlii/gym/service/implementation/TraineeServiceImpl.java

```diff
@@ -0,0 +1,321 @@
+        user.setLastName(updateRequestDTO.getLastName());
+        user.setIsActive(updateRequestDTO.getIsActive());
+        userRepo.save(user);
+
+        Trainee trainee = traineeRepo.findByUsername(updateRequestDTO.getUsername());
+        if (trainee == null) {
+            log.error("[{}] Trainee not found with username: {}", transactionId, updateRequestDTO.getUsername());
+            throw new EntityNotFoundException("Trainee not found with username: " + updateRequestDTO.getUsername());
+        }
+        trainee.setBirthday(updateRequestDTO.getBirthday());
+        trainee.setAddress(updateRequestDTO.getAddress());
+        traineeRepo.save(trainee);
+
+        List<TrainerDTO> trainers = trainee.getTrainers().stream().map(trainer -> {
+            TrainerDTO dto = new TrainerDTO();
+            dto.setUsername(trainer.getUser().getUsername());
+            dto.setFirstName(trainer.getUser().getFirstName());
+            dto.setLastName(trainer.getUser().getLastName());
+            TrainingTypeDTO specializationDTO = new TrainingTypeDTO(
+                    trainer.getSpecialization().getId(),
+                    trainer.getSpecialization().getTrainingTypeName()
+            );
+            dto.setSpecialization(specializationDTO);
+            return dto;
+        }).toList();
+
+        log.info("[{}] Profile successfully updated for username: {}", transactionId, user.getUsername());
+        return new UpdateResponseDTO(
+                user.getUsername(),
+                user.getFirstName(),
+                user.getLastName(),
+                trainee.getBirthday(),
+                trainee.getAddress(),
+                user.getIsActive(),
+                trainers
+        );
+    }
+
+    @Override
+    @Transactional
+    public void delete(String username) {
+        String transactionId = UUID.randomUUID().toString();
+        log.info("[{}] Initiating deletion for username: {}", transactionId, username);
+
+        Trainee trainee = traineeRepo.findByUsername(username);
+        if (trainee == null) {
+            log.error("[{}] Trainee does not exist for username: {}", transactionId, username);
+            throw new EntityNotFoundException("Trainee does not exist for username: " + username);
+        }
+
+        User user = userRepo.findByUsername(username);
+        if (user == null) {
+            log.error("[{}] User does not exist for username: {}", transactionId, username);
+            throw new EntityNotFoundException("User does not exist for username: " + username);
+        }
+        Set<Training> trainings = trainee.getTrainings();
+        for (Training t : trainings) {
+            trainingRepository.delete(t);
+            log.debug("[{}] Deleted training with ID: {} for username: {}", transactionId, t.getId(), username);
+        }
+        traineeRepo.delete(trainee);
+        log.info("[{}] Deleted trainee with username: {}", transactionId, username);
+
+        userRepo.delete(user);
```

Comment:
There shouldn't be many delete operations, so it should be configured with BO entity and deletion of Trainee object should make correspondent removal for User record from database.